server/src/unifyfs_server.c

```diff
@@ -675,6 +678,11 @@ static int unifyfs_exit(void)
     LOGDBG("stopping rpc service");
     margo_server_rpc_finalize();
 
+    ret = ABT_mutex_free(&app_configs_abt_sync);
```

Comment:
Not sure about this, but I think if `margo_finalize()` is being called as part of `margo_server_rpc_finalize()` above, and we're relying on Margo to init/fini Argobots, then we may not be able to safely call any ABT_xxx methods after that.